jdbc/src/main/java/org/jclouds/jdbc/strategy/JdbcStorageStrategy.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.jdbc.strategy;
+
+import autovalue.shaded.com.google.common.common.collect.ImmutableList;
+import com.google.inject.persist.Transactional;
+import org.jclouds.blobstore.LocalStorageStrategy;
+import org.jclouds.blobstore.domain.Blob;
+import org.jclouds.blobstore.domain.BlobAccess;
+import org.jclouds.blobstore.domain.BlobBuilder;
+import org.jclouds.blobstore.domain.ContainerAccess;
+import org.jclouds.blobstore.domain.MutableStorageMetadata;
+import org.jclouds.blobstore.domain.StorageMetadata;
+import org.jclouds.blobstore.domain.StorageType;
+import org.jclouds.blobstore.domain.internal.MutableStorageMetadataImpl;
+import org.jclouds.blobstore.options.CreateContainerOptions;
+import org.jclouds.blobstore.options.ListContainerOptions;
+import org.jclouds.domain.Location;
+import org.jclouds.jdbc.entity.Container;
+import org.jclouds.jdbc.predicates.validators.JdbcBlobKeyValidator;
+import org.jclouds.jdbc.predicates.validators.JdbcContainerNameValidator;
+import org.jclouds.jdbc.repository.ContainerRepository;
+import org.jclouds.logging.Logger;
+
+import javax.annotation.Resource;
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.persistence.NoResultException;
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * JdbcStorageStrategy implements a blob store that stores objects
+ * on a jdbc supported database. Content metadata and user attributes are stored in
+ * the database as well.
+ */
+public class JdbcStorageStrategy implements LocalStorageStrategy {
+
+   @Resource
+   protected Logger logger = Logger.NULL;
+
+   protected final Provider<BlobBuilder> blobBuilders;
+   protected final ContainerRepository containerRepository;
+   protected final JdbcContainerNameValidator jdbcContainerNameValidator;
+   protected final JdbcBlobKeyValidator jdbcBlobKeyValidator;
+
+   @Inject
+   protected JdbcStorageStrategy(Provider<BlobBuilder> blobBuilders,
```

Comment:
Remove the `protected` modifier, so you know the constructor is only called by the injector. Then you can safely remove all redundant `checkNotNull` checks.